client/me/developer/main.jsx

```diff
@@ -13,21 +14,116 @@ import twoStepAuthorization from 'calypso/lib/two-step-authorization';
 import withFormBase from 'calypso/me/form-base/with-form-base';
 import ReauthRequired from 'calypso/me/reauth-required';
 import { recordGoogleEvent } from 'calypso/state/analytics/actions';
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+import { successNotice, removeNotice } from 'calypso/state/notices/actions';
 import { isFetchingUserSettings } from 'calypso/state/user-settings/selectors';
 import { DeveloperFeatures } from './features/index';
 import { getIAmDeveloperCopy } from './get-i-am-a-developer-copy';
 
 import './style.scss';
 
 class Developer extends Component {
-	toggleIsDevAccount = ( isDevAccount ) => {
+	developerSurveyNoticeId = 'developer-survey';
+
+	getSurveyHref = () => {
+		return 'handle_me';
+	};
+
+	setDeveloperSurveyCookie = ( value, maxAge ) => {
+		document.cookie = cookie.serialize( 'developer_survey', value, {
+			path: '/',
+			maxAge,
+		} );
+	};
+
+	shouldShowDevSurveyNotice = ( isDevAccount ) => {
+		const cookies = cookie.parse( document.cookie );
+
+		const isDevAccountEnabled = isDevAccount ?? this.props.getSetting( 'is_dev_account' );
+		const isEN = this.props.currentUser.localeSlug === 'en';
+
+		return (
+			isEN &&
+			isDevAccountEnabled &&
+			! [ 'completed', 'dismissed' ].includes( cookies.developer_survey )
+		);
+	};
+
+	showDevSurveyNotice = () => {
+		const noticeMessage = this.props.translate(
+			"Hey developer! How do {{i}}you{{/i}} use WordPress.com? Spare a moment? We'd love to hear what you think in a {{surveyLink}}quick survey{{/surveyLink}}.",
+			{
+				components: {
+					i: <i />,
+					surveyLink: (
+						<a
+							href={ this.getSurveyHref() }
+							target="_blank"
+							rel="noopener noreferrer"
+							onClick={ () => {
+								recordTracksEvent( 'calypso_me_developer_survey_clicked' );
+
+								this.props.removeNotice( this.developerSurveyNoticeId );
+							} }
+						/>
+					),
+				},
+			}
+		);
+
+		const noticeProps = {
+			id: this.developerSurveyNoticeId,
+			isPersistent: true,
+			onDismissClick: () => {
+				recordTracksEvent( 'calypso_me_developer_survey_dismissed' );
+
+				this.setDeveloperSurveyCookie( 'dismissed', 24 * 60 * 60 ); // 1 day
+
+				this.props.removeNotice( this.developerSurveyNoticeId );
+			},
+		};
+
+		this.props.successNotice( noticeMessage, noticeProps );
+		recordTracksEvent( 'calypso_me_developer_survey_impression' );
+	};
+
+	handleToggleIsDevAccount = ( isDevAccount ) => {
 		this.props.setUserSetting( 'is_dev_account', isDevAccount );
 
 		recordTracksEvent( 'calypso_me_is_dev_account_toggled', {
 			enabled: isDevAccount ? 1 : 0,
 		} );
+
+		if ( this.shouldShowDevSurveyNotice( isDevAccount ) ) {
+			this.showDevSurveyNotice();
+		} else {
+			this.props.removeNotice( this.developerSurveyNoticeId );
+		}
+
+		setTimeout( () => this.props.removeNotice( 'save-user-settings' ), 5000 );
```

Comment:
```suggestion
		setTimeout( () => this.props.removeNotice( 'save-user-settings' ), 3000 );
```
I know we agreed on `5000`, but seeing it in action, I feel we can probably shorten it to `3000`, especially when we're also auto-closing the "Thank you for your feedback!" notice in 3 seconds. What do you think?

Reply:
I don't have strong arguments and obviously there is no right objective decision, so I am ok with 3000 :)